runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingSideInputFetcher.java

```diff
@@ -320,7 +320,10 @@ private <SideWindowT extends BoundedWindow> Windmill.GlobalDataRequest buildGlob
                 .build())
         .setExistenceWatermarkDeadline(
             WindmillTimeUtils.harnessToWindmillTimestamp(
-                sideWindowStrategy.getTrigger().getWatermarkThatGuaranteesFiring(sideInputWindow)))
+                sideWindowStrategy
```

Comment:
I think that triggers cannot actually give a tight bound. You either have to change _all_ of them to return `maxTimestamp + allowedLateness` or just take the minimum here. I would suggest keeping it simple and doing the minimum here. In fact, I would even support deleting the method from `Trigger` and doing all of the logic here, since it is really just a one-off special case for AfterWatermark / DefaultTrigger